### PR TITLE
feat: operators 24h balance diff and new sync participation calc

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

**Review by dgusakov — APPROVED** [review]

File: src/storage/clickhouse/clickhouse.service.ts [hunk 1]
```diff
@@ -254,15 +257,21 @@ export class ClickhouseService implements OnModuleInit {
...
+          await ws.writeRow(
+            `(${slotTime}, '${pubKey}', '${p.validator_index || ''}', ${last_slot_of_epoch}, ${p.epoch_participation_percent}, ` +
+              `${otherAvg}, ${keysIndexed.get(pubKey)?.operatorIndex ?? 'NULL'}, '${keysIndexed.get(pubKey)?.operatorName || 'NULL'}')`,
+          );
+        } else {
+          notUserPercents.push(p.epoch_participation_percent);
+        }
       }
       await this.retry(async () => await ws.exec());
+      return notUserPercents.reduce((a, b) => a + b, 0) / notUserPercents.length;
```
Comment (madlabman) [inline src/storage/clickhouse/clickhouse.service.ts:274]: Possible division by zero

File: src/inspector/processing/stats-processing.service.ts [hunk 2]
```diff
@@ -64,9 +64,13 @@ export class StatsProcessingService implements OnModuleInit {
     });
 
     // Validator Sync Committee participation
-    const syncParticipationAvgPercents = await this.storage.getSyncParticipationAvgPercents(slot);
-    this.prometheus.userSyncParticipationAvgPercent.set(syncParticipationAvgPercents.user ?? 0);
-    this.prometheus.chainSyncParticipationAvgPercent.set(syncParticipationAvgPercents.chain ?? 0);
+    const syncUserParticipationAvgPercents = await this.storage.getUserSyncParticipationAvgPercents(slot);
+    this.prometheus.userSyncParticipationAvgPercent.set(syncUserParticipationAvgPercents.avg_percent ?? 0);
```
Comment (madlabman) [inline src/inspector/processing/stats-processing.service.ts:68]:
```
this.prometheus.userSyncParticipationAvgPercent
                syncUserParticipationAvgPercents.avg_percent
```

image

Comment (vgorkavenko) [reply src/inspector/processing/stats-processing.service.ts]: fixed

File: src/storage/clickhouse/clickhouse.service.ts (hunk 1 above)
Comment (vgorkavenko) [reply src/storage/clickhouse/clickhouse.service.ts]: It's an unlikely case. Handled

**Pull request merged**